### PR TITLE
`flutter/lib/src/`: refactoring if-chains into switch expressions

### PR DESCRIPTION
**Pull request by nate-thegrate**
Based on issue #144903, this pull request aims to bring the codebase more in line with the [Flutter repo style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values):

> ### Avoid using `if` chains or `?:` or `==` with enum values

<br>

```dart
// before
bool get shown {
  if (overlayEntry == null) {
    return false;
  }
  if (animationController != null) {
    return animationController!.status == AnimationStatus.completed ||
        animationController!.status == AnimationStatus.forward;
  }
  return true;
}


// after
bool get shown {
  return overlayEntry != null && switch (animationController?.status) {
    AnimationStatus.forward || AnimationStatus.completed || null => true,
    AnimationStatus.reverse || AnimationStatus.dismissed => false,
  };
}
```

This pull request is similar to [a previous PR with the same name](https://github.com/flutter/flutter/pull/146293) but has a bit more thorough refactoring overall.

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/search_anchor.dart [hunk 1]
```diff
@@ -1444,7 +1437,7 @@ class _SearchBarState extends State<SearchBar> {
                         ),
                       ),
                     ),
-                    if (trailing != null) ...trailing,
+                    ...?trailing,
```
Comment (justinmc) [inline packages/flutter/lib/src/material/search_anchor.dart:1440]: I didn't realize that was a thing :+1:

**Review by justinmc — APPROVED** [review]
LGTM :+1: Thanks for cleaning up the code as always.

@nate-thegrate Are you interested in getting contributor access by the way? I'd be happy to sponsor you. See https://github.com/flutter/flutter/wiki/Contributor-access.

**Comment by nate-thegrate** [comment]
> @nate-thegrate Are you interested in getting contributor access by the way? I'd be happy to sponsor you.

@justinmc That would be awesome, thank you very much!

**Pull request merged**